File: src/main/java/com/lessmarkup/userinterface/model/structure/MenuItemModel.java
package com.lessmarkup.userinterface.model.structure;

public class MenuItemModel {
    private final String url;
    private final String title;
    private final String imageUrl;
    private final boolean selected;
    private final int level;
    
    public MenuItemModel(String url, String title, String imageUrl, boolean selected, int level) {
        this.url = url;
        this.title = title;
        this.imageUrl = imageUrl;
        this.selected = selected;
        this.level = level;
    }

    /**
     * @return the url
     */
    public String getUrl() {
        return url;
    }

    /**
     * @return the title
     */
    public String getTitle() {
        return title;
    }

    /**
     * @return the imageUrl
     */
    public String getImageUrl() {
        return imageUrl;
    }

    /**
     * @return the selected
     */
    public boolean isSelected() {
        return selected;
    }

    /**
     * @return the level
     */
    public int getLevel() {
        return level;
    }
    
    
}

File: src/main/scala/com/lessmarkup/userinterface/model/structure/MenuItemModel.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.userinterface.model.structure

class MenuItemModel(url: String, title: String, imageUrl: String, selected: Boolean, level: Int) {

  def getUrl: String = url

  def getTitle: String = title

  def getImageUrl: String = imageUrl

  def isSelected: Boolean = selected

  def getLevel: Int = level
}
